src/main/osd/osd_max7456.c

```diff
@@ -0,0 +1,230 @@
+        .exti_irqn = EXTI15_10_IRQn
+};
+#endif
+
+void osdHardwareApplyConfiguration(void)
+{
+    max7456_init(osdVideoConfig()->videoMode);
+
+    textScreen_t *max7456TextScreen = max7456_getTextScreen();
+    osdSetTextScreen(max7456TextScreen);
+}
+
+void osdHardwareInit(void)
+{
+    LED0_ON;
+    delay(500);
+    max7456_hardwareReset();
+    LED0_OFF;
+
+    osdHardwareApplyConfiguration();
+
+    max7456_extiConfigure(&max7456LOSExtiConfig, &max7456VSYNCExtiConfig, &max7456HSYNCExtiConfig);
+
+    NVIC_InitTypeDef NVIC_InitStructure;
+
+    NVIC_InitStructure.NVIC_IRQChannelCmd = ENABLE;
+
+    NVIC_InitStructure.NVIC_IRQChannel = max7456LOSExtiConfig.exti_irqn;
+    NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = NVIC_PRIORITY_BASE(NVIC_PRIO_OSD_LOS);
+    NVIC_InitStructure.NVIC_IRQChannelSubPriority = NVIC_PRIORITY_SUB(NVIC_PRIO_OSD_LOS);
+    NVIC_Init(&NVIC_InitStructure);
+
+    NVIC_InitStructure.NVIC_IRQChannel = max7456VSYNCExtiConfig.exti_irqn;
+    NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = NVIC_PRIORITY_BASE(NVIC_PRIO_OSD_VSYNC);
+    NVIC_InitStructure.NVIC_IRQChannelSubPriority = NVIC_PRIORITY_SUB(NVIC_PRIO_OSD_VSYNC);
+    NVIC_Init(&NVIC_InitStructure);
+
+    NVIC_InitStructure.NVIC_IRQChannel = max7456HSYNCExtiConfig.exti_irqn;
+    NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = NVIC_PRIORITY_BASE(NVIC_PRIO_OSD_HSYNC);
+    NVIC_InitStructure.NVIC_IRQChannelSubPriority = NVIC_PRIORITY_SUB(NVIC_PRIO_OSD_HSYNC);
+    NVIC_Init(&NVIC_InitStructure);
+
+    if (osdFontConfig()->fontVersion != FONT_VERSION) {
+        // before
+        max7456_showFont();
+        delay(5000); // give the user a chance to power off before changing
+
+        max7456_resetFont();
+
+        // after
+        max7456_showFont();
+        delay(5000); // give the user a chance to power off after changing
+
+        osdFontConfig()->fontVersion = FONT_VERSION;
+        writeEEPROM();
+
+        max7456_clearScreen();
+    }
+}
+
+void osdHardwareUpdate(void)
+{
+
+#if 0
+    debug[3] = max7456_readStatus();
+#endif
+
+    max7456_writeScreen(&osdTextScreen, textScreenBuffer);
+}
+
+void osdHardwareCheck(void)
+{
+    static int checkCount = 0;
+
+    checkCount++;
+
+    if (!max7456_isOSDEnabled()) {
+        max7456_init(osdVideoConfig()->videoMode);
+    }
+
+#ifdef FACTORY_TEST
+    if (checkCount == 10) {
+        max7456_init(osdVideoConfig()->videoMode);
+    }
+#endif
+
+    max7456_updateLOSState();
+}
+
+void osdHardwareDrawLogo(void)
+{
+    osdSetRawCharacterAtPosition(14, 5, FONT_CHARACTER_CF_LOGO_W3xH2__1x1);
+    osdSetRawCharacterAtPosition(15, 5, FONT_CHARACTER_CF_LOGO_W3xH2__1x2);
+    osdSetRawCharacterAtPosition(16, 5, FONT_CHARACTER_CF_LOGO_W3xH2__1x3);
+    osdSetRawCharacterAtPosition(14, 6, FONT_CHARACTER_CF_LOGO_W3xH2__2x1);
+    osdSetRawCharacterAtPosition(15, 6, FONT_CHARACTER_CF_LOGO_W3xH2__2x2);
+    osdSetRawCharacterAtPosition(16, 6, FONT_CHARACTER_CF_LOGO_W3xH2__2x3);
```

Comment:
i think yes, there will be multiple graphics objects.    a large 4 character warning triangle for low battery springs to mind.  yes, all rectangular.  since YAGNI applies here I will leave this for a later day and just use an array for now.